### ISSUE TITLE
How to add new filter lists to the built-in adblocker?

### ISSUE THREAD

**Issue by 3oheicrw** [snaildos/Fifo-Browser#103]
Or is there any way to know which filter lists are used by default? The built-in adblocker worked very well but it's pretty much unable to configure it other than turn it on or not. BTW, I got UbO installed and it seems to work (I disabled the built-in adblocker but ads still not show up), but it doesn't show the number of ads get blocked like on other browsers and I'm unable to configure it. Clicking on the extension's icon does nothing.

**Comment by snaildos** [comment]
The built in ad-blocker uses Cliqz, and again we are working on making the browser more flexible and usable, this will have to be planned _later._
Will get back to you regarding UbO.

**Comment by 3oheicrw** [comment]
The built-in adblocker works pretty good out of the box. UbO is not really needed now. I only add information about the current state of it on Fifo to clarify that there is no alternative to the built-in adblocker that works and the built-in adblocker will be better if it's configurable by the users. I don't need UbO.

**Comment by 3oheicrw** [comment]
Since it's based on Cliqz, this seems to be already possible:

https://github.com/ghostery/adblocker/blob/master/packages/adblocker-electron/README.md

**Comment by snaildos** [comment]
Yeah, it sounds easy but it really isn't. I'll be looking into a way to add this into version #116 but again this wont be changeable via UI and only via manually editing files, the UI will be released in a later version.

**Comment by 3oheicrw** [comment]
> Yeah, it sounds easy but it really isn't. I'll be looking into a way to add this into version #116 but again this wont be changeable via UI and only via manually editing files, the UI will be released in a later version.

Min browser doesn't support installing extensions like Fifo so they have to rely on their own. But, what about just trying to make Ublock Origin works and install it out of the box like LibreWolf? :wink:

**Comment by jiangzhenjerry** [comment]
@3oheicrw I also see that uBlock Origin doesn't show the number of ads blocked, but that's the only issue I encounter. Everything else works just fine to me. I can configure it and there is no YouTube ads or whatsoever.
<img width="1281" alt="0" src="https://user-images.githubusercontent.com/9311082/182898547-cf0d1707-7cc0-47e2-bf97-90a551c23916.png">
<img width="1281" alt="1" src="https://user-images.githubusercontent.com/9311082/182898556-75dbb807-339c-4f26-88d4-ae296c2f61df.png">

**Comment by snaildos** [comment]
> @3oheicrw I also see that uBlock Origin doesn't show the number of ads blocked, but that's the only issue I encounter. Everything else works just fine to me. I can configure it and there is no YouTube ad or whatsoever.
> <img width="1281" alt="0" src="https://user-images.githubusercontent.com/9311082/182898547-cf0d1707-7cc0-47e2-bf97-90a551c23916.png">
> <img width="1281" alt="1" src="https://user-images.githubusercontent.com/9311082/182898556-75dbb807-339c-4f26-88d4-ae296c2f61df.png">
> 
> 

Hey, thank you for updating. What version of Fifo are you running

**Comment by jiangzhenjerry** [comment]
I'm using [Fifo-1.2.1-x64.nsis.7z](https://github.com/snaildos/Fifo-Browser/releases/download/v1.2.1/Fifo-1.2.1-x64.nsis.7z).

**Comment by snaildos** [comment]
I am really excited into trying to integrate "Clear Ad Block Cache" and custom filter lists into Fifo. Will be attempting to do this soon. 😄

**Comment by shlwapi** [comment]
> > @3oheicrw I also see that uBlock Origin doesn't show the number of ads blocked, but that's the only issue I encounter. Everything else works just fine to me. I can configure it and there is no YouTube ad or whatsoever.
> > <img alt="0" width="1281" src="https://user-images.githubusercontent.com/9311082/182898547-cf0d1707-7cc0-47e2-bf97-90a551c23916.png">
> > <img alt="1" width="1281" src="https://user-images.githubusercontent.com/9311082/182898556-75dbb807-339c-4f26-88d4-ae296c2f61df.png">
> 
> Hey, thank you for updating. What version of Fifo are you running?

Are both of you sure it's actually UbO that working? Did he turned off the built-in adblocker on that particular site before testing with UbO? I don't think UbO works. It's the built-in adblocker that he forgot to turn off that caused the impression that UbO is working. BTW, did he try to add new filters (regional) on UbO? Does UbO could update/autoupdate the filter lists? There are too many things to question than just a simple screenshot that proved nothing.

p/s: I don't use this browser, just come across this repo because of suggestion by github.

**Comment by shlwapi** [comment]
BTW, look at the popup window of UbO. Do you see anything? Nothing was blocked on the site (youtube.com) at all. It's actually the built-in adblocker that doing the job, not UbO.

**Comment by jiangzhenjerry** [comment]
> There are too many things to question than just a simple screenshot that proved nothing.
> 
> p/s: I don't use this browser, just come across this repo because of suggestion by github.

You didn't need to say that you didn't use this browser, because it is very obvious to me - you didn't even know about the built-in adblocker icon, which is right there next to the uBO icon. In my screenshots, the built-in adblocker icon is clearly grayed out, which means the built-in adblocker disabled.

> BTW, look at the popup window of UbO. Do you see anything? Nothing was blocked on the site (youtube.com) at all. It's actually the built-in adblocker that doing the job, not UbO.

I already said it is a bug and that uBO is indeed doing its job. You know why I'm so sure about it? Just turn uBO off and the ads will be up again.

**Comment by snaildos** [comment]
The problem with adding these custom filters, is that I need to create a new filter.dat file, and load both the cache.dat (Default Adblocking List) and the filter.dat as well as make an entirely new UI just to add and delete filters. 😓 
I'll try my hardest after v1.2.2 gets released though.

**Comment by snaildos** [comment]
#119 and #132
Adblocker going to need a rewrite tbh.

**Comment by bprlhe** [comment]
@jiangzhenjerry You gave me false hope. I installed UbO but I can't open the setting page like you. Only after I uninstalled it and reinstall again I could open the setting page. I could update filter lists and add new filter lists. But the extension is very unreliable. Sometimes it works. Sometimes it doesn't. It also slowed down the loading of some pages. Sometimes these pages just keep loading. This doesn't happen with the built-in adblocker.

**Comment by bprlhe** [comment]
Let me tell you all I can even have Adguard extension installed and I could open the setting page of it right on first installation unlike UbO. With UbO only after you remove it and reinstall you could open the popup window when clicking on it icon and from there open the setting page. But Adguard extension also suffers the same limitation as UbO. Sometimes it works. Sometimes it doesn't.

**Comment by uddlrl** [comment]
I'm more lucky as UbO works correctly for me. On mail.ru and news.mail.ru all ads are blocked.

**Comment by dbgrgl** [comment]
I think I know the problem boys. I'm a long time Brave user. This also happens with Brave's adblocker. Please note that Brave's adblocker also uses UbO's filters. The same filters being used, but sometimes Brave's adblocker misses it but UbO could block it. Why? Because Brave's adblocker is not yet 100% compatible with UbO and there are some APIs not implemented. This improves over time. Now I would say Brave's adblocker is very competent and could replace UbO. I have disabled UbO and use only Brave's adblocker for a while. The same is true for this situation. UbO is using some APIs not implemented by Fifo extension layer, so some sets of ads are not blocked. The APIs used to implement blocking of these ads are not there. For example, banner ads on the site soha.vn still being shown. UbO on Chromium/Firefox and Brave's adblocker have no problems handle these ads.

**Comment by zpu2tw** [comment]
Hey guys I wanted to tell you that I could have AdBlocker Ultimate installed and working too. The problem is it seems unable to update filter lists.

**Comment by snaildos** [comment]
Thank you everyone for your stars, contribution and time to write these issues. I was never expecting ~10 stars overnight.
I really do appreciate it.


Regarding the adblocker.
I don't know the creator of UbO and I also don't know how the extension works. Building on what @zpu2tw said, I do agree, both AdBlocker Ultimate and UbO both suffer the same issue, saving settings. Saving extension settings does not work if its not local storage, It's super strange, but again its something that will need to be looked at for the #104 FEL rewrite. 
Thank you everyone once again for your contributions towards this project.

**Comment by qqmomr** [comment]
You should thank github. Somehow your project keeps being suggested to me even though I don't develop in JavaScript. All of the project suggested to me are C++ and only you is TypeScript.

**Comment by oor52ntn** [comment]
Not working for me. UbO is struggling to block the ads. It seems it first tried to keep the ads from loading with partial success. The ads are not able to shown but the areas taken by them are still there as blank space (note: it should be able to hide these areas too). The ads are not yet given up and somehow some of them managed to come back. UbO then trying to block these ads. This constant fight between UbO and the ads caused the page to keeps loading indefinitely.

**Comment by 3oheicrw** [comment]
> Let me tell you all I can even have Adguard extension installed

I could have Adguard installed too and I found each time Fifo starts Adguard initializes it filters database again. This means the data is simply not saved. As it doesn't work now I uninstalled it. I didn't check if the filters I added manually are still there or not though.

**Comment by pwd96m4a** [comment]
Let this thread back to focus on the built-in adblocker please. About the third party extensions, they worked when first installed but no longer work on the next start because their data are not stored, it get lost after you close the browser. This is a known bug. The built-in adblocker should be invested on rather than the third party extensions.

**Comment by snaildos** [comment]
> Let this thread back to focus on the built-in adblocker please. About the third party extensions, they worked when first installed but no longer work on the next start because their data are not stored, it get lost after you close the browser. This is a known bug. The built-in adblocker should be invested on rather than the third party extensions.

Contacted the author that made the electron extensions module. If he doesn't fix it, this will need to be forked into what is known as FElv3.

**Comment by snaildos** [comment]
I'll be modifying the Adblocker in v1.3.0 (next update).
